### PR TITLE
Cleaning up ARMI Docs

### PR DESCRIPTION
**Pull request by john-science**
## What is the change?

This PR does two things:

1. The User and Developer manuals are changed from being multiple RST files to just one.
2. A TON of headers had to be fix, because they were the wrong level.

## Why is the change being made?

This all stems from trying to import the ARMI docs RST files downstream in other projects and render them:

1. It was just too hard to get the User and Developer `toctree`s to render downstream, as they are already nested `toctree`s in ARMI.
2. A ton of headers in the ARMI docs were the wrong level. Sphinx was able to fix this for the ARMI docs, but not when things were imported in downsteam projects.

In actuality, the RST headers should go in this order:

* `#` with overline, for parts
* `*` with overline, for chapters
* `=` for sections
* `-` for subsections
* `^` for subsubsections
* `"` for paragraphs

---

## Checklist

- [X] This PR has only [one purpose or idea](https://terrapower.github.io/armi/developer/tooling.html#one-idea-one-pr).
- [X] [Tests](https://terrapower.github.io/armi/developer/tooling.html#test-it) have been added/updated to verify that the new/changed code works.

- [X] The code style follows [good practices](https://terrapower.github.io/armi/developer/standards_and_practices.html).
- [X] The commit message(s) follow [good practices](https://terrapower.github.io/armi/developer/tooling.html).

- [X] The [release notes](https://terrapower.github.io/armi/release/index.html) (location `doc/release/0.X.rst`) are up-to-date with any important changes.
- [X] The [documentation](https://terrapower.github.io/armi/developer/tooling.html#document-it) is still up-to-date in the `doc` folder.
- [X] If any [requirements](https://terrapower.github.io/armi/developer/tooling.html#watch-for-requirements) were affected, mention it in the [release notes](https://terrapower.github.io/armi/release/index.html).
- [X] The dependencies are still up-to-date in `pyproject.toml`.

### CODE REVIEW

File: doc/user/manual_data_access.rst [hunk 1]
```diff
@@ -8,20 +8,20 @@ in programmatically building and manipulating inputs and gathering detailed info
...
-* :doc:`Core Parameters </user/core_parameters_report>`
-* :doc:`Component Parameters </user/component_parameters_report>`
-* :doc:`Assembly Parameters </user/assembly_parameters_report>`
-* :doc:`Block Parameters </user/block_parameters_report>`
+* :doc:`Table of all global settings <../user/inputs>`
+* :doc:`Reactor Parameters <../user/reactor_parameters_report>`
+* :doc:`Core Parameters <../user/core_parameters_report>`
+* :doc:`Component Parameters <../user/component_parameters_report>`
+* :doc:`Assembly Parameters <../user/assembly_parameters_report>`
+* :doc:`Block Parameters <../user/block_parameters_report>`
```
Comment (keckler) [inline doc/user/manual_data_access.rst:20]: These links don't appear to be working when building the internal documentation.

File: doc/user/inputs.rst [hunk 2]
```diff
@@ -1,6 +1,418 @@
...
+
+This file is a YAML file that you can edit manually with a text editor or with the ARMI GUI.
+
+Here is an excerpt from a settings file:
+
+.. literalinclude:: ../../../armi/tests/armiRun.yaml
+    :language: yaml
+    :lines: 3-15
+
+A full listing of settings available in the framework may be found in the :doc:`Table of all global settings </user/inputs>`.
```
Comment (keckler) [inline doc/user/inputs.rst:55]: I think this is trying to link to itself? Regardless, the link doesn't show up when I build.

**Review by keckler — COMMENTED** [review]
A couple comments about links for you. Otherwise, it appears that everything is still there, so that's good!

File: doc/user/inputs.rst (hunk 2 above)
Comment (john-science) [reply doc/user/inputs.rst]: Fixed

File: doc/user/manual_data_access.rst (hunk 1 above)
Comment (john-science) [reply doc/user/manual_data_access.rst]: Okay, I fixed that using `:ref:` commands.

**Review by keckler — APPROVED** [review]

**Pull request merged**